File: 0004-median-of-two-sorted-arrays/solution.c
double findMedianSortedArrays(int* nums1, int nums1size, int* nums2, int nums2size) {
    int totalsize = nums1size + nums2size;
    int merged[totalsize];

    int i=0, j = 0, k=0;
    while (i < nums1size && j < nums2size) {
        if (nums1[i] < nums2[j]) {
            merged[k++] = nums1[i++];
        } else {
            merged[k++] = nums2[j++];
        }
    }

    while (i < nums1size) {
        merged[k++] = nums1[i++];
    }
    while (j < nums2size) {
        merged[k++] = nums2[j++];
    }

    if (totalsize % 2 != 0) {
        return (double)merged[totalsize / 2];
    } else {
        return (double)(merged[totalsize / 2 - 1] + merged[totalsize / 2]) / 2;
    }
}

File: 0004-median-of-two-sorted-arrays/solution.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int [] arr = new int[nums1.length + nums2.length];
        int j = 0;
        for(int i : nums1){
            arr[j] = i;
            j++;
        }
        
        for(int i : nums2){
            arr[j] = i;
            j++;
        }

        Arrays.sort(arr);
        double avg ;
        int len = arr.length;
        if(len%2 ==1 ){
            return (double) arr[len/2];
        }
        else{
            int mid1 = arr[len/2 - 1];
            int mid2 = arr[len/2];
            return ((double) mid1 + (double)mid2)/2.0;
        }
        
    
    // return Math.round(avg*100000.0)/100000.0;
    }
}
